event_scheduler/event_scheduler.py

```diff
@@ -78,15 +90,19 @@ def run(self):
         """
         # localize variable access to minimize overhead
         # and to improve thread safety
-        lock = self._lock
+        cv = self._cv
         q = self._queue
+        timer = self._timer
         delayfunc = self.delayfunc
         timefunc = self.timefunc
         pop = heapq.heappop
         while True:
-            with lock:
-                if not q:
-                    continue
+            with cv:
+                if not q or timer:
+                    cv.wait()
```

Comment:
sorry, I have to ask a bunch of questions since I don't fully understand, how is the wait duration determined?

Reply:
If you look line 117, I have a comment to explain that. I also have a comment on line 25 to describe the purpose of the condition variable and under what conditions it waits.

Reply:
gotchya